handlers/utils/compute_utils.go

```diff
@@ -0,0 +1,1186 @@
+			}
+		}
+	}
+	return &instance, &host
+}
+
+func IsInstanceActive(instance *model.Instance, host *model.Host) bool {
+	if IsNoOp(instance.Data) {
+		return true
+	}
+
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	return isRunning(client, container)
+}
+
+func IsNoOp(data map[string]interface{}) bool {
+	b, ok := GetFieldsIfExist(data, "processData", "containerNoOpEvent")
+	if ok {
+		return InterfaceToBool(b)
+	}
+	return false
+}
+
+func GetContainer(client *client.Client, instance *model.Instance, byAgent bool) *types.Container {
+	if instance == nil {
+		return nil
+	}
+
+	// First look for UUID label directly
+	args := filters.NewArgs()
+	args.Add("label", fmt.Sprintf("%s=%s", UUIDLabel, instance.UUID))
+	options := types.ContainerListOptions{All: true, Filter: args}
+	labeledContainers, err := client.ContainerList(context.Background(), options)
+	if err == nil && len(labeledContainers) > 0 {
+		return &labeledContainers[0]
+	}
+
+	// Nest look by UUID using fallback method
+	options = types.ContainerListOptions{All: true}
+	containerList, err := client.ContainerList(context.Background(), options)
+	if err != nil {
+		return nil
+	}
+	container := findFirst(containerList, func(c *types.Container) bool {
+		if getUUID(c) == instance.UUID {
+			return true
+		}
+		return false
+	})
+
+	if container != nil {
+		return container
+	}
+	if externalID := instance.ExternalID; externalID != "" {
+		container = findFirst(containerList, func(c *types.Container) bool {
+			return idFilter(externalID, c)
+		})
+	}
+
+	if container != nil {
+		return container
+	}
+
+	if agentID := instance.AgentID; byAgent {
+		container = findFirst(containerList, func(c *types.Container) bool {
+			return agentIDFilter(strconv.Itoa(agentID), c)
+		})
+	}
+
+	return container
+
+}
+
+func isRunning(client *client.Client, container *types.Container) bool {
+	if container == nil {
+		return false
+	}
+	inspect, err := client.ContainerInspect(context.Background(), container.ID)
+	if err == nil {
+		return inspect.State.Running
+	}
+	return false
+}
+
+func getUUID(container *types.Container) string {
+	uuid, err := container.Labels[UUIDLabel]
+	if err {
+		return uuid
+	}
+
+	names := container.Names
+	if names == nil {
+		return fmt.Sprintf("no-uuid-%s", container.ID)
+	}
+
+	if strings.HasPrefix(names[0], "/") {
+		return names[0][1:]
+	}
+	return names[0]
+}
+
+func findFirst(containers []types.Container, f func(*types.Container) bool) *types.Container {
+	for _, c := range containers {
+		if f(&c) {
+			return &c
+		}
+	}
+	return nil
+}
+
+func idFilter(id string, container *types.Container) bool {
+	return container.ID == id
+}
+
+func agentIDFilter(id string, container *types.Container) bool {
+	containerID, ok := container.Labels["io.rancher.container.agent_id"]
+	if ok {
+		return containerID == id
+	}
+	return false
+}
+
+func RecordState(client *client.Client, instance *model.Instance, dockerID string) {
```

Comment:
No because record state currently does something fundamentally important: the state file it records is then picked up by another process that use that file to know what IP address to inject into the container. It is critically important to the container's networking.